app/frontend/src/components/views/project-view/components/TaskInfo.tsx

```diff
@@ -19,6 +31,62 @@ interface Props {
  * @param Props.task - current task
  */
 const TaskInfo: React.FC<Props> = ({ projectTitle, task }) => {
+  const [statusesOptions, setStatusesOptions] = useState<DropdownItem[]>([]);
+  const [statuses, setStatuses] = useState<StatusType[]>([]);
+
+  useEffect(() => {
+    (async function () {
+      if (!task) {
+        return;
+      }
+      try {
+        const response = await getStatuses(task.projectId as string);
+
+        setStatuses(response.statuses);
+        const options = response.statuses.map(item => ({
+          label: item.label,
+          value: item._id,
+        }));
+
+        setStatusesOptions([EMPTY_STATUS_OPTION, ...options]);
+      } catch (e) {
+        console.error(e);
+      }
+    })();
+  }, [ task ]);
+
+  const onStatusChange = async (value: string|number|null|undefined): Promise<void> => {
+    if (!task) {
+      return;
+    }
+    try {
+      const prevStatus = statuses.find(status => status._id === task.statusId);
+
+      if (prevStatus) {
```

Comment:
could not understand this code. Why there are 2 `updateStatus()` requests one after another? Maybe it could be simplified?

Reply:
it was an attempt to implement behaviour described on the screenshot. 
When task status is changed 2 requests are performed: one to remove task id from previous status's tasks array, second -- to push task id to new status's tasks array. 
i know that boards view is gonna be in separate pull request, just wanted to simplify future db migrations

![image](https://user-images.githubusercontent.com/31101125/151879763-96a5fb72-25f9-4927-a905-7780a8db7cab.png)

Reply:
anyway it does not seem like a good idea to perform 3 requests in a row

Reply:
removed redundant requests